### PR TITLE
refactor: feat: ! Refactoring and describing the lockpickDoor function moved to client/functions.lua

### PR DESCRIPTION
**Pull request by ArturMichalak**
## Description

The lockpickDoor, attemptPoliceAlert, isBlacklistedWeapon, isBlacklistedVehicle functions and their helper functions have been moved to clients/functions.lua

These can be used in main.lua using import, [inspired by ox_target](https://github.com/overextended/ox_target/blob/main/client/utils.lua)
```lua
local functions = require 'client.functions'
local lockpickDoor, attemptPoliceAlert, isBlacklistedWeapon, isBlacklistedVehicle in functions
```

Some conditions have been added to enable the usage of the lock pick.

When opened, the distance between the vehicle and the player's ped gets checked.

Previously, opening the lock on trucks was done in the middle of the vehicle. Now you just have to approach the driver's door.

Used statebag to hold information about the lock.

## Checklist

- [X] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [X] My pull request fits the contribution guidelines & code conventions.

BREAKING CHANGE:
In the statebag of the vehicle there is a saved information on whether it has been opened with a lock pick.

### CODE REVIEW

**Review by Manason — APPROVED** [review]
Skimmed through things. Seems like an improvement.

**Comment by Manason** [comment]
Looks like there are some conflicts. Should have probably merged yours first. Sorry :(

**Comment by ArturMichalak** [comment]
I will resolve conflicts soon

File: shared/functions.lua [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
----@param coord2 vector3[] The second set of coordinates.
----@param distance number The maximum allowed distance for them to be considered close.
-function IsCloseToCoords(coord1, coord2, distance)
+--- @param coord1 vector3[] The first set of coordinates.
+--- @param coord2 vector3[] The second set of coordinates.
+--- @param distance number The maximum allowed distance for them to be considered close.
+--- @return boolean true if the distance between two entities is less than the distance parameter.
+function public.isCloseToCoords(coord1, coord2, distance)
     return #(coord1 - coord2) < distance
-end
```
Comment (Manason) [inline shared/functions.lua:7]: This seems like an unnecessary abstraction

File: fxmanifest.lua [hunk 2]
```diff
@@ -1,34 +1,37 @@
...
-repository 'https://github.com/Qbox-project/qbx_vehiclekeys'
+--[[ Resource Information ]]--
 version '1.0.0'
+license 'GPL-3.0-or-later'
+description 'vehicle key management system'
+repository 'https://github.com/Qbox-project/qbx_vehiclekeys'
+
+--[[ Manifest ]]--
+dependencies {
+    'ox_lib',
```
Comment (Manason) [inline fxmanifest.lua:15]: This will make it so if you restart ox_lib, then this resource will be stopped. Is that the desired behavior?

Comment (ArturMichalak) [reply fxmanifest.lua:15]: There are a few lib.callbacks in the script, so without ox_lib it may throw errors

File: shared/functions.lua (hunk 1 above)
Comment (ArturMichalak) [reply shared/functions.lua:7]: This function was already in the code. It improves readability a little bit, which is why I started using it.

**Pull request merged**